### PR TITLE
[Feat](nereids) add pull up literal when infer predicates

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
This pr add pull up literal when infer predicates, we can infer tmp.col1='abc'  and tmp.col2 = 'def' from subquery tmp, pull up to inner join, and combined with join on condition, we can infer ds.col1='abc' and ds.col2='def'
```sql
select
	*
from
	(
	select
		tmp.col1,
		tmp.col2
	from
		(
		select
			'abc' as col1,
			'def' as col2
		from
			test_pull_up_predicate_literal
               ) tmp
               inner join test_pull_up_predicate_literal ds 
               on tmp.col1 = ds.col1 and tmp.col2 = ds.col2
	) t
where col1 = 'abc' and col2 = 'def';
```

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run external

**Review by 924060929 — APPROVED** [review]

File: regression-test/data/nereids_rules_p0/infer_predicate/pull_up_predicate_literal.out [hunk 1]
```diff
@@ -0,0 +1,3139 @@
+-- This file is automatically generated. You should know what you did if you want to edit this
+-- !test_pull_up_literal --
+PhysicalResultSink
+--PhysicalProject
+----hashJoin[INNER_JOIN] hashCondition=((col1 = ds.col1) and (col2 = ds.col2)) otherCondition=()
+------PhysicalProject
+--------PhysicalOneRowRelation
+------filter((ds.col1 = 'abc') and (ds.col1 = ds.col1) and (ds.col2 = 'def') and (ds.col2 = ds.col2))
```
Comment (morrySnow) [inline regression-test/data/nereids_rules_p0/infer_predicate/pull_up_predicate_literal.out:8]: `ds.col1 = ds.col1` is another bug in infer fiter?

**Review by morrySnow — APPROVED** [review]

**Pull request merged**